model/schema.json

```diff
@@ -647,99 +645,152 @@
         "api_key_id",
         "api_key"
       ]
-    }
-  },
-  "checkin": {
-    "title": "Checkin",
-    "description": "An Elastic Agent checkin to Fleet",
-    "type": "object",
-    "properties": {
-      "_id": {
-        "description": "The unique identifier for the Elastic Agent checkin",
-        "type": "string",
-        "format": "uuid"
-      },
-      "@timestamp": {
-        "description": "Date/time the checkin was created",
-        "type": "string",
-        "format": "date-time"
-      },
-      "agent": { "$ref": "#/definitions/agent-metadata" },
-      "host": { "$ref": "#/definitions/host-metadata" },
-      "server": { "$ref": "#/definitions/server-metadata" },
-      "status": {
-        "description": "The current overall status of the Elastic Agent",
-        "type": "string",
-        "enum": ["enrolling", "healthy", "error", "degraded", "offline", "unenrolling", "upgrading"]
-      },
-      "message": {
-        "description": "The current overall status message of the Elastic Agent",
-        "type": "string"
+    },
+
+    "checkin": {
```

Comment:
This was not scoped correctly (as defined in the draft-07 definition) so the go structs for checkin were not created. 

The openapi spec for checkin requests has [less strict definition](https://github.com/elastic/fleet-server/blob/main/model/openapi.yml#L340), @cmacknz should we update the openapi spec to check for the checkin as it would be defined here?